### PR TITLE
dependency maintenance

### PR DESCRIPTION
**Pull request by nickgros**
Ran `pnpm update --recursive` and followed up with a few manual version changes to achieve the following
- Fix all issues that appear in pnpm audit that have an upgrade path
- Fix all but one peer dependency issues
- Remove project-specific `resolutions` fields (these had no effect on dependency resolution)
- Small changes to account for package updates

Additionally, update our SBOM upload workflow to continue getting vulnerability alerts (still needed while we wait on https://github.com/dependabot/dependabot-core/issues/7434).

### CODE REVIEW

File: .github/workflows/sbom.yml [hunk 1]
```diff
@@ -28,6 +27,6 @@ jobs:
           name: sbom
           path: _manifest/spdx_2.2
       - name: SBOM upload
-        uses: jhutchings1/spdx-to-dependency-graph-action@v0.0.2
```
Comment (nickgros) [inline .github/workflows/sbom.yml:31]: original repo was archived and points to this new action

File: apps/SageAccountWeb/package.json [hunk 2]
```diff
@@ -103,14 +103,6 @@
...
     ]
   },
-  "resolutions": {
-    "js-yaml": "3.13.1",
-    "react": "18.2.0",
-    "react-hot-toast": "2.2.0",
-    "@types/react": "18.0.17",
-    "@types/react-dom": "18.0.6",
-    "minimatch": "^3.1.2"
-  },
```
Comment (nickgros) [inline apps/SageAccountWeb/package.json:113]: According to the output of `pnpm install`, these resolution entries are ignored. Only the base package.json `resolutions` are used.

Comment (nickgros) [reply apps/SageAccountWeb/package.json:113]: See docs for `overrides`: https://pnpm.io/package_json#pnpmoverrides  - this is the same feature, just a different property name.

File: packages/synapse-react-client/src/components/StandaloneQueryWrapper/StandaloneQueryWrapper.tsx [hunk 3]
```diff
@@ -46,6 +46,7 @@ type StandaloneQueryWrapperOwnProps = {
     | 'columnAliases'
     | 'noContentPlaceholderType'
     | 'showLastUpdatedOn'
+    | 'visibleColumnCount'
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/StandaloneQueryWrapper/StandaloneQueryWrapper.tsx:49]: This was missing in the prop signature, but the prop was still passed down, so this is a type-only change.

File: packages/synapse-react-client/src/synapse-queries/dataaccess/useAccessRequirements.ts [hunk 4]
```diff
@@ -158,6 +158,7 @@ export function useCreateLockAccessRequirement(
       if (options?.onSuccess) {
         return options.onSuccess(data, variables, ctx)
       }
+      return
```
Comment (nickgros) [inline packages/synapse-react-client/src/synapse-queries/dataaccess/useAccessRequirements.ts:161]: These functions must have a return statement. If I had to guess, it probably has something to do with [this TypeScript change](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-1.html) and some (outdated) react-query types.

File: packages/synapse-react-client/src/theme/useTheme.tsx [hunk 5]
```diff
@@ -6,7 +6,7 @@ import defaultMuiThemeOptions from './DefaultTheme'
 import type { PartialDeep } from 'type-fest'
 
 export function mergeTheme(
-  themeOverrides: PartialDeep<ThemeOptions>,
+  themeOverrides: ThemeOptions | PartialDeep<ThemeOptions>,
```
Comment (nickgros) [inline packages/synapse-react-client/src/theme/useTheme.tsx:9]: MUI/Emotion update required this change.

File: packages/vite-config/src/vite-config.ts
Comment (nickgros) [inline packages/vite-config/src/vite-config.ts:1]: code formatting

File: apps/SageAccountWeb/package.json [hunk 6]
```diff
@@ -5,48 +5,47 @@
...
     "react-dom": "18.2.0",
-    "react-easy-crop": "^4.2.0",
-    "react-query": "3.39.1",
-    "react-router-dom": "5.3.3",
-    "react-spinners": "^0.11.0",
-    "react-use": "^17.3.1",
-    "sass": "^1.30.0",
+    "react-easy-crop": "^4.7.5",
+    "react-query": "3.39.3",
+    "react-router-dom": "^5.3.4",
```
Comment (nickgros) [inline apps/SageAccountWeb/package.json:22]: semver change: `react-router-dom`: `5.3.3 -> ^5.3.4` 

We were already using 5.3.4 in a few places. This fixes a peer dependency mismatch issue.

File: packages/synapse-react-client/test/utils/hooks/SynapseAPI/useEntityBundle.test.tsx [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import { renderHook } from '@testing-library/react-hooks'
+import { renderHook } from '@testing-library/react'
```
Comment (nickgros) [inline packages/synapse-react-client/test/utils/hooks/SynapseAPI/useEntityBundle.test.tsx:1]: Functionality in @testing-library/react-hooks has been ported to @testing-library/react, so we dropped @testing-library/react-hooks

File: packages/synapse-react-client/test/utils/hooks/usePreFetchResource.integration.test.tsx [hunk 8]
```diff
@@ -36,14 +36,13 @@ describe('usePreFetchResource tests', () => {
       ),
     )
 
-    const { result, waitForNextUpdate } = renderHook(() =>
```
Comment (nickgros) [inline packages/synapse-react-client/test/utils/hooks/usePreFetchResource.integration.test.tsx:39]: waitForNextUpdate is not in @testing-library/react, but that's ok, we can just use `waitFor`

File: packages/synapse-react-client/src/assets/illustrations/challenges_illustration.svg
Comment (nickgros) [inline packages/synapse-react-client/src/assets/illustrations/challenges_illustration.svg:1]: Had an `<?xml >` tag that was not parseable by upgraded esbuild

File: packages/synapse-react-client/test/utils/hooks/SynapseAPI/useEntityBundle.test.tsx [hunk 9]
```diff
@@ -25,22 +13,17 @@ const expected: EntityBundle = {
   },
 }
 
-const SynapseClient = require('../../../../src/synapse-client/SynapseClient')
-SynapseClient.getEntityBundleV2 = jest.fn().mockResolvedValue(expected)
```
Comment (nickgros) [inline packages/synapse-react-client/test/utils/hooks/SynapseAPI/useEntityBundle.test.tsx:29]: Replaced dynamic `require` import with `jest.spyOn`

File: packages/synapse-react-client/test/utils/hooks/SynapseAPI/useEntityBundle.test.tsx [hunk 10]
```diff
@@ -1,21 +1,9 @@
...
-import { QueryClient } from 'react-query'
+import { renderHook, waitFor } from '@testing-library/react'
 import useGetEntityBundle from '../../../../src/synapse-queries/entity/useEntityBundle'
 import { EntityBundle } from '@sage-bionetworks/synapse-types'
 import { MOCK_CONTEXT_VALUE } from '../../../../mocks/MockSynapseContext'
-import FullContextProvider from '../../../../src/utils/context/FullContextProvider'
-
-const queryClient = new QueryClient()
-
-const wrapper = (props: { children: React.ReactChildren }) => (
```
Comment (nickgros) [inline packages/synapse-react-client/test/utils/hooks/SynapseAPI/useEntityBundle.test.tsx:11]: Replace with existing utility

File: packages/synapse-react-client/test/utils/hooks/SynapseAPI/useEntityBundle.test.tsx [hunk 11]
```diff
@@ -25,22 +13,17 @@ const expected: EntityBundle = {
...
   it('correctly calls SynapseClient', async () => {
     const entityId = 'syn123'
 
-    const { result, waitFor } = renderHook(() => useGetEntityBundle(entityId), {
-      wrapper,
+    const { result } = renderHook(() => useGetEntityBundle(entityId), {
+      wrapper: createWrapper(),
     })
 
-    await waitFor(() => result.current.isSuccess)
```
Comment (nickgros) [inline packages/synapse-react-client/test/utils/hooks/SynapseAPI/useEntityBundle.test.tsx:43]: The returned `waitFor` must have allowed returning a boolean. The imported `waitFor` check doesn't work unless an error is thrown, such as if an `expect` fails.

File: packages/vite-config/src/vitest-config.ts
Comment (nickgros) [inline packages/vite-config/src/vitest-config.ts:1]: Config itself unchanged, just reformatted and structure changed to work with type changes

File: .github/workflows/sbom.yml
Comment (nickgros) [inline .github/workflows/sbom.yml:1]: This workflow pushes a dependency list from our lockfile to a Microsoft tool that generates and reports security vulnerabilities. It's our workaround to get vulnerability alerts until GitHub supports PNPM.

File: apps/SageAccountWeb/package.json [hunk 12]
```diff
@@ -56,27 +55,27 @@
...
     "timers-browserify": "^2.0.12",
-    "tslib": "^2.3.1",
-    "type-fest": "^3.7.2",
-    "typescript": "5.0.4",
-    "url": "^0.11.0",
-    "util": "^0.12.4",
-    "vite": "^4.3.4",
+    "tslib": "^2.6.0",
+    "type-fest": "^3.13.0",
+    "typescript": "5.1.6",
```
Comment (nickgros) [inline apps/SageAccountWeb/package.json:73]: Upgraded TS everywhere

File: apps/synapse-oauth-signin/package.json [hunk 13]
```diff
@@ -4,50 +4,49 @@
...
-    "plotly.js-basic-dist": "^2.22.0",
+    "plotly.js-basic-dist": "^2.24.3",
     "react": "^18.2.0",
-    "react-bootstrap": "^1.5.2",
+    "react-bootstrap": "^1.6.7",
     "react-cookie": "4.0.0",
     "react-dom": "^18.2.0",
-    "react-query": "3.39.1",
-    "react-router-dom": "5.3.3",
-    "react-spinners": "^0.11.0",
```
Comment (nickgros) [inline apps/synapse-oauth-signin/package.json:22]: react-spinners also unused here

File: apps/synapse-oauth-signin/package.json [hunk 14]
```diff
@@ -57,29 +56,29 @@
...
-    "markdown-it-synapse": "^1.1.10",
+    "markdown-it-synapse": "workspace:^",
     "markdown-it-synapse-heading": "^1.0.1",
-    "markdown-it-synapse-math": "^3.0.4",
+    "markdown-it-synapse-math": "^3.0.5",
     "markdown-it-synapse-table": "^1.0.6",
-    "memfs": "^3.4.7",
-    "msw": "^0.49.3",
+    "memfs": "^3.5.3",
+    "msw": "^1.2.2",
```
Comment (nickgros) [inline apps/synapse-oauth-signin/package.json:64]: Bump to match other packages

File: packages/synapse-react-client/test/utils/hooks/SynapseAPI/useGetEntityChildren.test.tsx
Comment (nickgros) [inline packages/synapse-react-client/test/utils/hooks/SynapseAPI/useGetEntityChildren.test.tsx:1]: Renamed to .ts

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**